### ISSUE TITLE
Define OrientationSensor interfaces

### ISSUE THREAD

**Issue by alexshalamov** [w3c/orientation-sensor#15]
The OrientationSensor interface could define methods and internal slots (quaternion), while subclasses can explicitly define model and what kind of low-level sensors are used for fusion, e.g.:

- AbsoluteOrientationSensor : OrientationSensor, uses gyroscope, acceleromter and magnetometer.
- GeomagneticOrientationSensor : OrientationSensor, uses accelerometer and magnetometer.
- RelativeOrientationSensor : OrientationSensor, uses gyroscope and accelerometer.

**Comment by pozdnyakov** [comment]
SGTM

**Comment by kenchris** [comment]
Can we actually implement this. I mean on Android the above "AbsoluteOrientationSensor" might use different impl if magnetometer isnt available. Is it possible for us to detect that?

**Comment by kenchris** [comment]
Apart from that, I think this is fine.

**Comment by anssiko** [comment]
@kenchris, it seems your explainer doc does identify high value use cases for all three:
 
https://w3c.github.io/motion-sensors/#orientation
https://w3c.github.io/motion-sensors/#geomagnetic-orientation
https://w3c.github.io/motion-sensors/#relative-orientation

That said, the use case for Relative Orientation Sensor ("This avoids the delay which low and high pass filters introduce.") could be more explicit.

**Comment by alexshalamov** [comment]
> Can we actually implement this. I mean on Android the above "AbsoluteOrientationSensor" might use different impl if magnetometer isnt available. Is it possible for us to detect that?

@kenchris Yes, we can always check what low level sensors are supported by the platform.

**Comment by pozdnyakov** [comment]
@kenchris in that perspective (and looks like we all agreed :) ) do you think it's worth renaming `Orientation Sensor`  into `Absolute Orientation Sensor` in https://w3c.github.io/motion-sensors/#orientation ?

**Comment by kenchris** [comment]
If we can do that without instantiating them, awesome, then let's go with this :-)

**Comment by kenchris** [comment]
@pozdnyakov yes that makes sense.

**Comment by pozdnyakov** [comment]
> If we can do that without instantiating them, awesome, then let's go with this :-)
we can :D

**Comment by alexshalamov** [comment]
Reopening the issue, as it should be closed only when last GeomagneticOrientationSensor is specified.

**Comment by anssiko** [comment]
Marked as "Level 2" per WG call decision.
